src/librustc_lint/context.rs

```diff
@@ -473,150 +474,121 @@ pub trait LintContext: Sized {
     fn sess(&self) -> &Session;
     fn lints(&self) -> &LintStore;
 
-    fn lookup_and_emit<S: Into<MultiSpan>>(&self, lint: &'static Lint, span: Option<S>, msg: &str) {
-        self.lookup(lint, span, msg).emit();
-    }
-
-    fn lookup_and_emit_with_diagnostics<S: Into<MultiSpan>>(
+    fn lookup_with_diagnostics<S: Into<MultiSpan>>(
         &self,
         lint: &'static Lint,
         span: Option<S>,
```

Comment:
While you're at it, `span: Option<impl Into<MultiSpan>>,` might read a bit nicer.